agogosml/agogosml/common/kafka_streaming_client.py

```diff
@@ -58,22 +61,35 @@ def __init__(self, config):  # pragma: no cover
             signal.signal(signal.SIGTERM, self.exit_gracefully)
 
     @staticmethod
-    def create_kafka_config(user_config: dict) -> dict:  # pragma: no cover
+    def create_kafka_config(user_config: dict, logger: Logger) -> dict:
         """Create the kafka configuration."""
+        if not user_config.get("KAFKA_ADDRESS"):
+            raise ValueError("KAFKA_ADDRESS is not set in the config object.")
+
+        base_config = user_config.get("KAFKA_CONFIG") or {}
+        if base_config and isinstance(base_config, str):
+            try:
+                base_config = json.loads(base_config)
+            except ValueError:
+                logger.warning("Could not parse Kafka Config provided as a string. Expecting JSON format.")
+                base_config = {}
+
         config = {
+            **base_config,
             "bootstrap.servers": user_config.get("KAFKA_ADDRESS"),
             "enable.auto.commit": False,
             "auto.offset.reset": "latest",
-            "default.topic.config": {'auto.offset.reset': 'latest'},
+            "default.topic.config": {'auto.offset.reset': 'latest'}
         }
 
         if user_config.get('KAFKA_CONSUMER_GROUP') is not None:
             config['group.id'] = user_config['KAFKA_CONSUMER_GROUP']
 
         if user_config.get('KAFKA_DEBUG') is not None:
             config['debug'] = user_config['KAFKA_DEBUG']
+            logger.debug("Kafka Debug set to: {}".format(config['debug']))
 
-        if user_config.get('EVENTHUB_KAFKA_CONNECTION_STRING'):
+        if user_config.get('EVENTHUB_KAFKA_CONNECTION_STRING') is not None:
```

Comment:
here we have to ensure that user_config.get('EVENTHUB_KAFKA_CONNECTION_STRING') won't return an empty string if it's not set